verification-annotations/src/tests.rs

```diff
@@ -85,6 +86,110 @@ fn t5() {
     verifier::assert_ne!(a, a+1);
 }
 
+#[cfg_attr(not(feature = "verifier-crux"), test)]
+#[cfg_attr(feature = "verifier-crux", crux_test)]
+fn bytes() {
+    let a = verifier::verifier_nondet_bytes(8);
+    for i in a.iter() {
+        verifier::assume(*i == 42);
+    }
+    if verifier::is_replay() {
+        println!("{:?}", a);
+    }
+    verifier::assert_eq!(a.len(), 8);
+    verifier::assert_ne!(a[2], 0u8);
+    verifier::assert_eq!(a[3], 42u8);
+}
+
+#[cfg_attr(not(feature = "verifier-crux"), test)]
+#[cfg_attr(feature = "verifier-crux", crux_test)]
+fn cstring() {
+    let a = verifier::verifier_nondet_cstring(8);
+
+    if verifier::is_replay() {
+        println!("{:?}", a);
+    }
+
+    // force string to be plain ASCII - to keep things simple
+    for i in a.as_bytes().iter() {
+        verifier::assume(i.is_ascii_alphabetic());
+    }
+
+    for i in a.as_bytes().iter() {
```

Comment:
Do you need the `.iter()` here? `for` does that for you. You can also do all this in one assertion:
```
verifier::assert!(a.as_bytes().iter().all(char::is_ascii))
```

Reply:
I'm dropping the .iter() here and in the other place that I used '.as_bytes().iter()' a few lines above.

But I find that this point-free style less readable so I prefer the for loop.